scripts/docker_push.sh

```diff
@@ -1,19 +1,21 @@
 #!/bin/bash
 
 # Dockerhub login
-[ ! -z "$DOCKER_PASSWORD" ] || [ ! -z "$DOCKER_USERNAME" ];
-  echo "DOCKER_USERNAME and DOCKER_PASSWORD must be set as environment variables!";
-  exit 1;
-echo "$DOCKER_PASSWORD" | docker login -u "$DOCKER_USERNAME" --password-stdin
-
-# if the build is for a pushed tag, use that tag for the Docker image;
-# otherwise, use "latest"
-if [[ -z "$TRAVIS_TAG" ]]; then
-  image_release_tag="latest";
+if [[ ! -n "$DOCKER_PASSWORD" || ! -n "$DOCKER_USERNAME" ]] >&2; then
```

Comment:
`>&2` should go after `echo`.  To my knowledge `[[` doesn't output anything to stdout.